lib/rex/post/meterpreter/extensions/stdapi/fs/file.rb

```diff
@@ -75,7 +75,7 @@ class << self
   #
   # Raises a RequestError if +root+ is not a directory.
   #
-  def File.search( root=nil, glob="*.*", recurse=true, timeout=-1 )
+  def File.search( root=nil, glob="*.*", recurse=true, startdate="default_startDate", enddate="default_endDate", timeout=-1)
```

Comment:
Note that in Ruby these aren't keyword arguments, and are just positional arguments. I believe adding additional arguments in the middle is a backwards incompatible change. We'd need to either update these modules, or I believe `startdate` will accidentally be given the `timeout` value
https://github.com/rapid7/metasploit-framework/blob/d9e4aadc40f6429154245aa85cf155484386ac67/modules/post/windows/gather/enum_files.rb#L60-L71

Or we'll want to update this line instead to use named arguments
```suggestion
  def File.search( root=nil, glob="*.*", recurse=true, timeout=-1, startdate: "default_startDate", enddate: "default_endDate")
```

And the caller would use:
```
files += client.fs.file.search(root, glob, recurse, startdate: startDate, enddate: endDate)
```

Also a useful read: https://thoughtbot.com/blog/ruby-2-keyword-arguments

Reply:
Thanks super useful link. I initially moved my new arguments before timeout since timeout isn't used and left default and I didn't know enough Ruby to manage positional vs. keyword arguments.  Your suggestion is cleaner and I'll update.

Reply:
The default value should also be nil to imply that there is no start/end date filter to be applied instead of special string values.

Reply:
I adjusted as suggested.